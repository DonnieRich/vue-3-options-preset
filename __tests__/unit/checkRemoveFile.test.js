const { vol } = require("memfs");
jest.mock("fs/promises");

const { removeFile } = require("../../src/bin/removeFile");

// import all json config files
const { config } = require('../../src/config/config.test');
const { BASE_DIR } = config.get();

describe(removeFile, () => {
    beforeEach(() => {
        vol.reset();
    });

    it("Should pass if the file style.css has been removed", async () => {

        vol.fromJSON(
            {
                [`${BASE_DIR}/style.css`]: "my style"
            },
            "/"
        );

        await removeFile(BASE_DIR, '.css');
        expect(vol.toJSON()).toMatchSnapshot();
    });

    it("Should pass if all .css files have been removed", async () => {

        vol.fromJSON(
            {
                [`${BASE_DIR}/styles/style.css`]: "my style",
                [`${BASE_DIR}/styles/base.css`]: "my base style",
            },
            "/"
        );

        await removeFile(`${BASE_DIR}/styles`, '.css');
        expect(vol.toJSON()).toMatchSnapshot();
    });

    it("Should pass if the file main.js has not been removed", async () => {

        const json = {
            [`${BASE_DIR}/main.js`]: "my js"
        };
        vol.fromJSON(
            json,
            "/"
        );

        await removeFile(BASE_DIR, '.css');
        expect(vol.toJSON()).toEqual(json);
    });

    it("Should pass if the file Test.vue has been removed", async () => {

        const json = {
            [`${BASE_DIR}/Test.vue`]: "my vue"
        };
        vol.fromJSON(
            json,
            "/"
        );

        await removeFile(BASE_DIR, '.vue');
        expect(vol.toJSON()).toMatchSnapshot();
    });

    it("Should pass if the file App.vue has not been removed", async () => {

        const json = {
            [`${BASE_DIR}/Test.vue`]: "my vue",
            [`${BASE_DIR}/App.vue`]: "my App vue"
        };
        vol.fromJSON(
            json,
            "/"
        );

        await removeFile(BASE_DIR, '.vue');
        expect(vol.toJSON()).toMatchSnapshot();
    });
});