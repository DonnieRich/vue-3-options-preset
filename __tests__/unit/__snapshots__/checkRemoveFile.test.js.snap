// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`removeFile Should pass if all .css files have been removed 1`] = `
{
  "/src/styles": null,
}
`;

exports[`removeFile Should pass if the file App.vue has not been removed 1`] = `
{
  "/src/App.vue": "my App vue",
}
`;

exports[`removeFile Should pass if the file Test.vue has been removed 1`] = `
{
  "/src": null,
}
`;

exports[`removeFile Should pass if the file style.css has been removed 1`] = `
{
  "/src": null,
}
`;
